src/javascript/crypto/e2e/otr/integration_test.html

```diff
@@ -0,0 +1,168 @@
+
+  var storageRemote = createStorage();
+  stubs.setPath('goog.storage.mechanism.HTML5LocalStorage', function() {
+    return storageRemote;
+  });
+  var remoteKey = {
+    p: goog.crypt.hexToByteArray('fd7f53811d75122952df4a9c2eece4e7f611b7523ce' +
+    'f4400c31e3f80b6512669455d402251fb593d8d58fabfc5f5ba30f6cb9b556cd7813b801' +
+    'd346ff26660b76b9950a5a49f9fe8047b1022c24fbba9d7feb7c61bf83b57e7c6a8a6150' +
+    'f04fb83f6d3c51ec3023554135a169132f675f3ae2b61d72aeff22203199dd14801c7'),
+    q: goog.crypt.hexToByteArray('9760508f15230bccb292b982a2eb840bf0581cf5'),
+    g: goog.crypt.hexToByteArray('f7e1a085d69b3ddecbbcab5c36b857b97994afbbfa3' +
+    'aea82f9574c0b3d0782675159578ebad4594fe67107108180b449167123e84c281613b7c' +
+    'f09328cc8a6e13c167a8b547c8d28e0a3ae1e2bb3a675916ea37f0bfa213562f1fb627a0' +
+    '1243bcca4f1bea8519089a883dfe15ae59f06928b665e807b552564014c3bfecf492a'),
+    x: goog.crypt.hexToByteArray('6cc79680373c932e88827f06af036a5dfd4de65d'),
+    y: goog.crypt.hexToByteArray('d25e344833a2541c159fc2e98a74ba119c60ae13d2f' +
+    'bf6ac505d25c38f82886b40422c44c43ce11404d664dc75c4124ae16cda56919a9dbc75b' +
+    '705d71b615d8542290654705d0c9e909b0729a28face053f6726a110ab90e8f39c37e452' +
+    'e2341d356a0555262df80dfa8caa420abc4d67edad0d25ee3cb8d1917e633761997eb')
+  };
+  remoteContext = createContext(remoteKey);
+  remoteSession = new e2e.otr.Session(remoteContext, new Uint8Array([2,2,2,2]));
+}
+
+function tearDown() {
+  stubs.reset();
+}
+
+var createStorage = function() {
+  var storage = {};
+  var get = function(key) {
+    return storage[key];
+  };
+  var set = function(key, value) {
+    storage[key] = value;
+  };
+  return {get: get, set: set};
+};
+
+function createContext(key) {
+  var storage = goog.storage.mechanism.HTML5LocalStorage();
+  storage.set('PRIVKEY', JSON.stringify(key));
+  storage.set('PUBKEY', JSON.stringify(new e2e.otr.pubkey.Dsa(key).pack()));
+  return new e2e.otr.ContextImpl();
+};
+
+function testAKE() {
+  var lastSent;
+  stubs.setPath('e2e.otr.Session.prototype.send', function(data) {
+    console.log(data);
+    if (data instanceof e2e.otr.message.Message) {
+      data = data.prepareSend();
+    }
+    lastSent = data;
+  });
+
+  assertEquals(MSGSTATE.PLAINTEXT, localSession.getMsgState());
+  assertEquals(AUTHSTATE.NONE, localSession.getAuthState());
+  assertEquals(MSGSTATE.PLAINTEXT, remoteSession.getMsgState());
+  assertEquals(AUTHSTATE.NONE, remoteSession.getAuthState());
+
+  // Start OTR by sending DH_COMMIT to remote.
+  localSession.initiateOtr();
+
+  assertEquals(MSGSTATE.PLAINTEXT, localSession.getMsgState());
+  assertEquals(AUTHSTATE.AWAITING_DHKEY, localSession.getAuthState());
+  assertEquals(MSGSTATE.PLAINTEXT, remoteSession.getMsgState());
+  assertEquals(AUTHSTATE.NONE, remoteSession.getAuthState());
+
+  // Remote receives receives DH_COMMIT, sends back DH_KEY.
+  remoteSession.processMessage(lastSent);
```

Comment:
I'm getting 'Message type not supported.' for a type=2 (DH_COMMIT). In this file I see there's a goog.require('e2e.otr.message.DhCommit') and dhcommit.js does e2e.otr.message.handler.add(e2e.otr.message.DhCommit)